File: Valid Compressed String.java
static int checkCompressed(String S, String T) {
        // code here
        
        int i=0,j=0;
        while(j<T.length()){
            if(Character.isLetter(T.charAt(j))){
                if(i>=S.length() || T.charAt(j)!=S.charAt(i)){
                    return 0;
                }
                i++;
                
            }else{
                int count = Character.getNumericValue(T.charAt(j));
                while(j+1<T.length() && Character.isDigit(T.charAt(j+1))){
                    count = count*10+Character.getNumericValue(T.charAt(j+1));
                    j++;
                }
                if(count>S.length()-i){
                    return 0;
                }
                i=i+count;
            }
            j++;
        }
        
        if(i==S.length()){
            return 1;
        }else return 0;
        
    }

File: Valid Compressed String.cpp
class Solution {

  public:

    int checkCompressed(string S, string T) {

        // code herewhile

        int i = 0, j = 0;

        while (i < S.size() && j < T.size())

        {

         

            string sum = "0";

            while (T[j] >= '0' && T[j] <= '9')

            {

                sum +=T[j];

                j++;

            }

            i += stoi(sum);

     

            if (S[i] != T[j])

                return 0;

            else if (S[i] == T[j] and i < S.size() and j < T.size())

            {

                i++;

                j++;

            }

        }

        if (i == S.size() && j == T.size())

            return 1;

        return 0;

    }

};
